lib/spack/spack/test/util/binary_resource.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import os
+import shutil
+
+import pytest
+
+import spack.util.binary_resource as br
+import spack.util.executable
+from spack.paths import spack_root
+
+datadir = os.path.join(spack_root, "lib", "spack", "spack", "test", "data", "binary_resource")
+file_file = os.path.join(datadir, "file.bat")
+
+pytestmark = pytest.mark.only_windows("only runs on windows")
+
+
+@pytest.fixture
+def mock_binary_resource_root(monkeypatch, tmpdir):
+    resource_root = tmpdir.mkdir("binary-resources")
+
+    def _binary_resource_root():
+        return resource_root
+
+    monkeypatch.setattr(spack.util.binary_resource, "binary_resource_root", _binary_resource_root)
+
+
+@pytest.fixture
+def no_system(working_env):
+    """Fixture preventing system binaries from being detected"""
+    os.environ["PATH"] = ""
+
+
+@pytest.fixture
+def tar_dir():
+    return os.path.dirname(spack.util.executable.which("tar").path)
+
+
+@pytest.fixture
+def on_system(working_env):
+    """Fixture explicityly adding binary resource to PATH"""
+    env = spack.util.environment.EnvironmentModifications()
+    env.append_path("PATH", datadir)
+    env.apply_modifications()
+
+
+def test_ensure_or_acquire_no_acquire(mock_binary_resource_root, config, no_system, monkeypatch):
+    def _fake_acquire(self):
+        os.makedirs(br.binary_resource_root() / "file" / "bin")
+        shutil.copy(file_file, br.binary_resource_root() / "file" / "bin")
+        return True
+
+    monkeypatch.setattr(
+        spack.util.binary_resource.BinaryResource, "acquire_resource", _fake_acquire
+    )
+    br.win_ensure_or_acquire_resource("file")
+    file = spack.util.executable.which("file")
+    assert file
+    assert "file-5.4.1 magicfile from /usr/share/bin" in file(output=str)
+
+
+@pytest.mark.skipif(not spack.util.executable.which("tar"), reason="tar required; not available")
+def test_ensure_or_acquire_acquire_resource(mock_binary_resource_root, config, tar_dir, no_system):
+    os.environ["PATH"] = tar_dir
+    br.win_ensure_or_acquire_resource("file")
+    file = spack.util.executable.which("file")
+    assert file
+    assert "file-5.4.1 magicfile from /usr/share/bin" in file(output=str)
+
+
+def test_ensure_or_acquire_resource_on_system(
+    mock_binary_resource_root, mutable_config, on_system
+):
+    # if resource not on system, ensure the resource acquisition fails with an exception
```

Comment:
If I understand, you meant "I've set this up so that if `file` is not on the system, that this will fail". FWIW I think the next test covers this sufficiently, and it does not depend on the availability of `file`. If you want to cover this, I'd recommend updating the next test to add a fake `file` binary and then re-execute after the first (expected) failure.

Reply:
My intent here was to explicitly test the short circuiting logic I have that avoids fetching any resources when file _is_ present on the system, where the failure here would indicate an issue with that logic. 
I'll consolidate as suggested